src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -818,8 +818,10 @@ bool RegionNode::optimize_trichotomy(PhaseIterGVN* igvn) {
   } else if (cmp1->Opcode() == Op_CmpF || cmp1->Opcode() == Op_CmpD ||
```

Comment:
This table is getting rather large. Time to add an abstraction on cmp-nodes?

Reply:
Thanks for reviewing this. The rationale for the existing ones is described here:

https://mail.openjdk.java.net/pipermail/hotspot-compiler-dev/2018-October/030928.html

The reasons for excluding floating point number, pointer comparison and SubTypeCheck don't seem to all fall under a single common justification that could be abstracted away,